File: ObjC_Example/Native/Views/NativeAdCollectionView.m
//
//  NativeAdCollectionView.m
//  ObjC_Example
//
//  Copyright (c) 2015年 FAN Communications. All rights reserved.
//

#import "NativeAdCollectionView.h"

@interface NativeAdCollectionView ()

@property (nonatomic, weak) IBOutlet UILabel *prTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *shortTextLabel;
@property (nonatomic, weak) IBOutlet UIImageView *adImageView;

@end

@implementation NativeAdCollectionView

#pragma mark - NADNativeViewRendering.h

- (UILabel *)prTextLabel
{
    return _prTextLabel;
}

- (UILabel *)shortTextLabel
{
    return _shortTextLabel;
}

- (UIImageView *)adImageView
{
    return _adImageView;
}

@end

File: Swift_Example/Native/Views/NativeAdCollectionView.swift
//
//  NativeAdCollectionView.swift
//  ObjC_Example
//
//  Copyright © 2015年 FAN Communications. All rights reserved.
//

import UIKit
import NendAd

class NativeAdCollectionView: UICollectionViewCell, NADNativeViewRendering {
 
    @IBOutlet fileprivate weak var nativeAdPrTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdShortTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdImageView: UIImageView!
    
    // MARK: - NADNativeViewRendering
    
    func prTextLabel() -> UILabel! {
        return self.nativeAdPrTextLabel
    }
    
    func shortTextLabel() -> UILabel! {
        return self.nativeAdShortTextLabel
    }
    
    func adImageView() -> UIImageView! {
        return self.nativeAdImageView
    }    
}
